src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -45,7 +45,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     </Otherwise>
   </Choose>
 
-  <Target Name="_CheckRuntimeIdentifier" BeforeTargets="_CheckForInvalidConfigurationAndPlatform">
+  <!-- TODO: change BeforeTargets="PrepareForBuild" back to BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
+             once we update to a version of MSBuild where _CheckForInvalidConfigurationAndPlatform is not an initial target -->
+  <!-- With _CheckForInvalidConfigurationAndPlatform as an initial target, this can stop a NuGet restore from succeeding, which we don't want -->
+  <Target Name="_CheckRuntimeIdentifier" BeforeTargets="PrepareForBuild">
```

Comment:
> I'm also curious what this has to do with this change. Can you split it out?

When the .NET SDK comes from a NuGet package, this targets file is not imported during the NuGet restore operation.  Switching to an SDK means the targets file is always included, which means if `_CheckForInvalidConfigurationAndPlatform` is an initial target, then `_CheckRuntimeIdentifier` can cause NuGet restore to fail.

I can't update to the latest CLI due to dotnet/cli#4884.  @piotrpMSFT, what build number could I use to get a recent version of the .NET CLI from before dotnet/cli#4824 went in?